crates/holochain_state/src/mutations.rs

```diff
@@ -102,16 +90,49 @@ pub fn insert_record_scratch(
     }
 }
 
-/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the database.
-pub fn insert_op(txn: &mut Transaction, op: &DhtOpHashed) -> StateMutationResult<()> {
-    insert_op_when(txn, op, Timestamp::now())
+/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the Authored database.
+pub fn insert_op_authored(
+    txn: &mut Ta<DbKindAuthored>,
+    op: &DhtOpHashed,
+) -> StateMutationResult<()> {
+    insert_op_when(txn, op, Timestamp::now(), None)
+}
+
+/// Insert a [`DhtOp`](holochain_types::dht_op::DhtOp) into the DHT database.
+///
+/// If `transfer_data` is None, that means that the Op was locally validated
+/// and is being included in the DHT by self-authority
+pub fn insert_op_dht(
+    txn: &mut Ta<DbKindDht>,
+    op: &DhtOpHashed,
+    transfer_data: Option<(AgentPubKey, TransferMethod, Timestamp)>,
```

Comment:
It's confusing that this is mixed in, I've been wondering what this was about the whole way down

Reply:
It is known at the moment the op is inserted, so it seems to make sense to include it with the insertion

Reply:
I think this would be clearer as a custom `enum` instead of adding meaning to `None`.

Reply:
@cdunster in the database a None is represented as NULL, would you extend the custom value to the database too, i.e. make the field non-null and respect a special sentinel value that represents the self-authority case with no transfer data? Or just make a new enum that causes NULL to be written instead of data, for the self-authority variant?